### PR TITLE
SED-2750 timeseries   more button shows already existing filters

### PR DESCRIPTION
**Pull request by lucianbaciu**

### CODE REVIEW

File: projects/step-frontend/src/lib/modules/timeseries/components/dashboard/dashboard.component.ts [hunk 1]
```diff
@@ -131,12 +129,17 @@ export class DashboardComponent implements OnInit, OnDestroy {
...
+        const apiFilter = FilterUtils.convertToApiFilterItem(item);
+        apiFilter.removable = false; // make all the fields not removable once saved
+        return apiFilter;
+      }) || [];
     this._dashboardService.saveDashboard(this.dashboard).subscribe((response) => {});
   }
 
   addDashlet(metric: MetricType) {
-    this.dashboard.dashlets.push({
+    let newDashlet: DashboardItem = {
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/components/dashboard/dashboard.component.ts:142]:
```suggestion
    const newDashlet: DashboardItem = {
```

File: projects/step-frontend/src/lib/modules/timeseries/modules/filter-bar/components/dashboard-filter-bar/dashboard-filter-bar.component.ts [hunk 2]
```diff
@@ -120,6 +121,10 @@ export class DashboardFilterBarComponent implements OnInit, OnDestroy {
   private _matDialog = inject(MatDialog);
   private _snackbar = inject(MatSnackBar);
 
+  trackByAttributeFn = (index: number, item: FilterBarItem): string => {
+    return item.attributeName;
+  };
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/modules/filter-bar/components/dashboard-filter-bar/dashboard-filter-bar.component.ts:126]: Where is it used?

**Review by dvladir — CHANGES_REQUESTED** [review]

File: projects/step-frontend/src/lib/modules/timeseries/modules/filter-bar/components/dashboard-filter-bar/dashboard-filter-bar.component.ts [hunk 3]
```diff
@@ -133,6 +138,21 @@ export class DashboardFilterBarComponent implements OnInit, OnDestroy {
     }
     this._internalFilters = this.context.getFilteringSettings().filterItems;
 
+    this.context.onAttributesChange().subscribe((attributes: Record<string, MetricAttribute>) => {
```
[the remark this answers is not in the document]
Comment (lucianbaciu) [reply projects/step-frontend/src/lib/modules/timeseries/modules/filter-bar/components/dashboard-filter-bar/dashboard-filter-bar.component.ts:141]: yes, good catch

**Comment by lucianbaciu** [comment]
@neogucky please check

**Review by dvladir — APPROVED** [review]

**Review by neogucky — APPROVED** [review]

**Pull request merged**